src/main/java/uk/gov/homeoffice/digital/sas/timecard/kafka/producers/KafkaProducerService.java

```diff
@@ -31,33 +31,42 @@ public KafkaProducerService(
   public void sendMessage(String messageKey, Class<T> resourceType,
                           T resource, KafkaAction action) {
     var kafkaEventMessage = new KafkaEventMessage<>(projectVersion, resourceType, resource, action);
-    ListenableFuture<SendResult<String, KafkaEventMessage<T>>> future =
-        kafkaTemplate.send(
-            topicName,
-            messageKey,
-            kafkaEventMessage
-        );
-
-    listenableFutureReporting(resource, kafkaEventMessage, future);
+    CompletableFuture<SendResult<String, KafkaEventMessage<T>>> future = null;
+    try {
+      future = kafkaTemplate.send(
+          topicName,
+          messageKey,
+          kafkaEventMessage
+      );
+      completeKafkaTransaction(future);
+      logKafkaMessage(messageKey, kafkaEventMessage, future);
+    } catch (ExecutionException | InterruptedException e) {
+      if  (Thread.interrupted()) {
+        Thread.currentThread().interrupt();
+      }
+      log.error(String.format(
+          "Message with key [ %s ] failed sending to topic [ %s ] action [ %s ]",
```

Comment:
Use constant for the string literal. It can then be used in the tests as well

Reply:
Actioned